File: php/rna-transcription/rna-transcription.php
<?php

function toRna(string $s) : string {
  return join(array_map("transcribe", str_split($s)));
}

function transcribe(string $nucleotide) : string {
  switch ($nucleotide) {
  case 'G': return 'C';
  case 'C': return 'G';
  case 'T': return 'A';
  case 'A': return 'U';
  default: throw new Exception;
  }
}

File: csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class RnaTranscription
{
    static Dictionary<char, char> dna_to_rna = new Dictionary<char, char> {
        {'G', 'C'},
        {'C', 'G'},
        {'T', 'A'},
        {'A', 'U'}
    };

    public static string ToRna(string nucleotide)
    {
        return String.Join(null, nucleotide.Select(c => dna_to_rna[c]));
    }
}
